packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/objects/ObjectDefinitionBuilder.kt

```diff
@@ -321,6 +326,28 @@ open class ObjectDefinitionBuilder {
     eventsDefinition = EventsDefinition(events)
   }
 
+  inline fun <reified T> Events() where T : Enumerable, T : Enum<T> {
+    val primaryConstructor = T::class.primaryConstructor
```

Comment:
is that correct to use enumerable based events will come with some startup cost from reflection?

Reply:
Yes, it'll, but the cost should be negligible because enums classes are generally small and easy to work with.